saleor/graphql/checkout/mutations/checkout_complete.py

```diff
@@ -164,116 +174,260 @@ def validate_checkout_addresses(
-                            "Some of the checkout lines variants are unavailable.",
-                            code=CheckoutErrorCode.UNAVAILABLE_VARIANT_IN_CHANNEL.value,
-                            params={"variants": not_available_variants_ids},
-                        )
-                    }
-                )
-            if not lines:
-                raise ValidationError(
-                    {
-                        "lines": ValidationError(
-                            "Cannot complete checkout without lines.",
-                            code=CheckoutErrorCode.NO_LINES.value,
-                        )
-                    }
-                )
-            discounts = load_discounts(info.context)
-            checkout_info = fetch_checkout_info(checkout, lines, discounts, manager)
+            )
+        if not lines:
+            raise ValidationError(
+                {
+                    "lines": ValidationError(
+                        "Cannot complete checkout without lines.",
+                        code=CheckoutErrorCode.NO_LINES.value,
+                    )
+                }
+            )
+        discounts = load_discounts(info.context)
+        checkout_info = fetch_checkout_info(checkout, lines, discounts, manager)
 
-            cls.validate_checkout_addresses(checkout_info, lines)
+        cls.validate_checkout_addresses(checkout_info, lines)
 
-            requestor = get_user_or_app_from_context(info.context)
-            if requestor and requestor.has_perm(AccountPermissions.IMPERSONATE_USER):
-                # Allow impersonating user and process a checkout by using user details
-                # assigned to checkout.
-                customer = checkout.user
-            else:
-                customer = info.context.user
+        requestor = get_user_or_app_from_context(info.context)
+        if requestor and requestor.has_perm(AccountPermissions.IMPERSONATE_USER):
+            # Allow impersonating user and process a checkout by using user details
+            # assigned to checkout.
+            customer = checkout.user
+        else:
+            customer = info.context.user
 
-            site = load_site(info.context)
-            order, action_required, action_data = complete_checkout(
-                manager=manager,
-                checkout_info=checkout_info,
-                lines=lines,
-                payment_data=data.get("payment_data", {}),
+        result["checkout_info"] = checkout_info
+        result["lines"] = lines
+        result["discounts"] = discounts
+        result["customer"] = customer
+        result["manager"] = manager
+        result["site"] = load_site(info.context)
+        result["tracking_code"] = analytics.get_client_id(info.context)
+
+        return result
+
+    @classmethod
+    def perform_mutation(
+        cls, _root, info, store_source, checkout_id=None, token=None, id=None, **data
+    ):
+        with transaction_with_commit_on_errors():
```

Comment:
I am not convinced that the main body of complete_checkout logic should placed in perform_mutation body. 
The main task mutation should be validation of the data, and calling bussiness logic (previously `complete_checkout`). 
I think that `complete_checkout` function should do the logic that is placed currently in the `perform_mutation`.